flipperdevices/flipperzero-firmware
# [FL-3486,FL-3392] fbt: various improvements and bug fixes

# What's new

- [FL-3486], [FL-3392] reworked `fbt flash` to autodetect swd probe
- also works for `ufbt`
- added `SWD_TRANSPORT` and `SWD_TRANSPORT_SERIAL` (formely `OPENOCD_ADAPTER_SERIAL`)
- github: added `appsymbols` artifact with debug .elfs for apps
- fbt: fixed missing regeneration for API cache in certain cases
- fbt, ufbt: added FAP_VERSION definition for app environments

# Verification 

- Build fw
- Run `fbt flash FORCE=1` with devboard in all modes & stlink
- Check out https://update.flipperzero.one/builds/firmware/hedger/fbtfixes/ for `appsymbols` artifact
- Try using FAP_VERSION in your external apps

# Checklist (For Reviewer)



[FL-3486]: https://flipperzero.atlassian.net/browse/FL-3486?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[FL-3392]: https://flipperzero.atlassian.net/browse/FL-3392?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
